channeldb/invoices.go

```diff
@@ -1311,6 +1319,14 @@ type InvoiceSlice struct {
 func (d *DB) QueryInvoices(q InvoiceQuery) (InvoiceSlice, error) {
 	var resp InvoiceSlice
 
+	// If MinCreationDate filter is greater than MaxCreationDate, we know
+	// there will be no matching results and can safely return early.
+	if q.MinCreationDate > 0 && q.MaxCreationDate > 0 {
```

Comment:
Since we check `!= 0` multiple times on this, you could have:

Reply:
Done.